Mnt 18 feature add http

Jade-Dragon88 opened the pull request:
Добавил сервис для api
Добавил сервис для http

Jade-Dragon88 committed:
+ apiService + httpService

Добавил в проект apiService и httpService

Изменил/добавил:
	modified:   src/components/pages/MentiList/MentiList.tsx
	new file:   src/services/api/api.service.ts
	new file:   src/services/api/http.service.ts

Jade-Dragon88 committed:
Корректный тип для Data в httpService

Изменил/добавил:
	modified:   src/services/api/http.service.ts

ivnvMkhl commented on src/services/api/http.service.ts:
```diff
@@ -0,0 +1,20 @@
+  readonly get = (url: string): Promise<Menti[]> => {
+    return fetch(url).then((response) => response.json());
+  };
+  readonly post = (url: string, data: Record<string, string>) => {
```
data переименовать в payload и типизируется он Any потомучто может быть и массивом и строкой и объектом любой вложенности

ivnvMkhl commented on src/services/api/api.service.ts:
```diff
@@ -0,0 +1,13 @@
+import { Menti } from '../../interfaces/menti.interfaces';
+import { httpService } from './http.service';
```
httpService не должен напрямую импортироваться он должен попасть в ApiService из коструктора

Jade-Dragon88 committed:
Скорректировал httpService
Выделил его в отдельный файл.

Добавил/изменил
	modified:   src/services/api/api.service.ts
	renamed:    src/services/api/http.service.ts -> src/services/http/http.service.ts
